kong/db/dao/init.lua

```diff
@@ -710,6 +712,10 @@ local function generate_foreign_key_methods(schema)
         end
 
         local entities, err
+        if options == nil or options.expand_shorthands ~= false then
```

Comment:
We need to check if we haven't disabled `options.expand_shorthands` ourselves by setting it to `false`. This can happen on `update` path since those methods first look for an entity in database. If one was found then they merge passed entity with the entity found polluting the record with shorthand_fields on update. So in order not to merge it with shorthands we need to check if we haven't disabled the expansion of them.

Reply:
Under what circumstance is it useful to have `shorthand_fields` be false?

Reply:
For example the method `check_update` performs it's operations in 2 steps:
1. select entity that's about to be updated
2. merge the received entity with the one found in database

On it's 1st step `check_update` uses a variation of DAO:`select` method. However if we do not disable shorthand expansion before selecting records from DB we'll get a DB record with shorthand fields expanded. Later on we'll merge it with data received via API and the shorthand fields that should only be returned on response to client will get inserted into database.